### PR TITLE
Show clear error on illegal config value access

### PR DESCRIPTION
**Pull request by voznesenskym**
Fixes https://github.com/pytorch/torchdynamo/issues/767

### CODE REVIEW

File: torchdynamo/config.py [hunk 1]
```diff
@@ -18,7 +19,7 @@
 # INFO print compiled functions + graphs
 # WARN print warnings (including graph breaks)
 # ERROR print exceptions (and what user code was being processed when it occurred)
-log_level = logging.WARNING
+log_level = logging.ERROR
```
Comment (voznesenskym) [inline torchdynamo/config.py:22]: Fun bonus to stem the spam

Comment (voznesenskym) [reply torchdynamo/config.py:22]: nevermind theres some cursed test

Comment (ezyang) [reply torchdynamo/config.py]: IMO warning should be on by default, it just needs to be less chatty

File: torchdynamo/config.py [hunk 2]
```diff
@@ -95,3 +96,33 @@
...
+        assert AccessLimitingConfig.config_has_attr(
+            name
+        ), f"Trying to get {name} - this value does not exist in torchdynamo.config"
+        return getattr(sys.modules[f"{__name__}.real"], name)
+
+    def __getattr__(self, name):
+        return self.config_get_attr(name)
+
+    def __setattr__(self, name, value):
+        self.config_set_attr(name, value)
```
Comment (ezyang) [inline torchdynamo/config.py:124]: You should factor this somewhere so torchinductor.config can get this treatment too

**Review by ezyang — APPROVED** [review]

**Comment by mlazos** [comment]
Could we make the config an object at the TorchDynamo module level
`config = TorchDynamoConfig()` and then override `setattr` normally?  It will probably have a lot fewer edge cases. I'm okay if this code works though.

**Review by mlazos — APPROVED** [review]

**Comment by voznesenskym** [comment]
> Could we make the config an object at the TorchDynamo module level `config = TorchDynamoConfig()` and then override `setattr` normally? It will probably have a lot fewer edge cases. I'm okay if this code works though.

That's a big change - I started with that, but it changed too many callsites. Patching sys modules felt like the best back-compat option.

**Comment by mlazos** [comment]
> > Could we make the config an object at the TorchDynamo module level `config = TorchDynamoConfig()` and then override `setattr` normally? It will probably have a lot fewer edge cases. I'm okay if this code works though.
> 
> That's a big change - I started with that, but it changed too many callsites. Patching sys modules felt like the best back-compat option.

What callsites does it change though? I think you'd have to remove the import? Is that what you mean? torchdynamo.config.x = y would still work I think?

**Comment by voznesenskym** [comment]
> > > Could we make the config an object at the TorchDynamo module level `config = TorchDynamoConfig()` and then override `setattr` normally? It will probably have a lot fewer edge cases. I'm okay if this code works though.
> > 
> > 
> > That's a big change - I started with that, but it changed too many callsites. Patching sys modules felt like the best back-compat option.
> 
> What callsites does it change though? I think you'd have to remove the import? Is that what you mean? torchdynamo.config.x = y would still work I think?

Hmm where would the class live? if in torchdynamo/config.py, it would have to be torchdynamo.config.config right? or youd have to instantiate an instance of a config?

**Comment by mlazos** [comment]
> > > > Could we make the config an object at the TorchDynamo module level `config = TorchDynamoConfig()` and then override `setattr` normally? It will probably have a lot fewer edge cases. I'm okay if this code works though.
> > > 
> > > 
> > > That's a big change - I started with that, but it changed too many callsites. Patching sys modules felt like the best back-compat option.
> > 
> > 
> > What callsites does it change though? I think you'd have to remove the import? Is that what you mean? torchdynamo.config.x = y would still work I think?
> 
> Hmm where would the class live? if in torchdynamo/config.py, it would have to be torchdynamo.config.config right? or youd have to instantiate an instance of a config?

it would live in root torchdynamo
and just be config = Config()

**Pull request merged**